### ISSUE TITLE
How to decode the ssl traffic with the certification file?

### ISSUE THREAD

**Issue by tianchao-haohan** [ntop/nDPI#638]
Is that possible to apply dpi on ssl traffic if given the certification file or private key?
And how to import the  certification file or private key if it is supported?

**Comment by kYroL01** [comment]
@tianchao-haohan Decode SSL/TLS traffic is not that easy, even if you are certificate or private key.
It depends on many factors, and on the cipher suites used. For example, if the traffic is decrypted with [Diffie-Hellman](https://en.wikipedia.org/wiki/Diffie%E2%80%93Hellman_key_exchange) is kinda impossibile to recontruct the keys needed for decryption.
A possible way is to decrypt traffic encrypted with RSA, but you have to obtain the Pre-Master Secret, then the Master Secret and then obtain the keys that, used a crypto library (i.e [Libgcrypt](https://github.com/gpg/libgcrypt) or or Openssl) rebuild the pair of keys for the final decrpytion.
Basically, you have to resolve the Simmetric and Asimmetric decryption.
in nDPI this does not possible.
If you want to know more, I suggest you to read 

1. https://www.cryptologie.net/article/340/tls-pre-master-secrets-and-master-secrets/
2. https://security.stackexchange.com/questions/20803/how-does-ssl-tls-work
3. http://hayageek.com/rsa-encryption-decryption-openssl-c/
